Remove misleading sentence

Hawk777 opened the pull request:
The sentence, “It inherits BufferedReader and BufferedWriter, and
further supports seek() and tell() functionality,” is misleading: it
suggests that BufferedRandom is special in that it is seekable, whereas
BufferedReader and BufferedWriter are not. However, this is not the
case: assuming the underlying raw stream is seekable (which is equally a
requirement of BufferedRandom), then BufferedReader and BufferedWriter
*are* seekable. Remove the misleading sentence.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->

Hawk777 committed:
Remove misleading sentence

The sentence, “It inherits BufferedReader and BufferedWriter, and
further supports seek() and tell() functionality,” is misleading: it
suggests that BufferedRandom is special in that it is seekable, whereas
BufferedReader and BufferedWriter are not. However, this is not the
case: assuming the underlying raw stream is seekable (which is equally a
requirement of BufferedRandom), then BufferedReader and BufferedWriter
*are* seekable. Remove the misleading sentence.

benjaminp reviewed (changes requested):
I think we still need to say something about `seek` & `tell` otherwise it's not clear what `BufferedRandom` is for at all unless you know what a "random" stream is. We could say `seek` and `tell` *must* be supported.

Hawk777 committed:
Indicate that seek and tell are mandatory

For BufferedReader and BufferedWriter, seek and tell operations are
optional (they may or may not exist based on the underlying stream). For
BufferedRandom, they are mandatory: a BufferedRandom should not be
constructed over an unseekable underlying stream. Document this.
